packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1525,61 +1525,63 @@ abstract class RenderObjectElement<T extends RenderObjectWidget> extends Buildab
     // 6. Sync null with any items in the list of keys that are still
```

Comment:
the comment above is wrong (in particular, it explicitly says it does the work backwards)